sips/SIP-20: Native Stake

```diff
@@ -0,0 +1,134 @@
+![hacker_attack](../assets/sip-tmp/hacker_attack.jpeg)
+Moreover, such [attacks](https://twitter.com/pikapikasui/status/1783025361275765136) have occurred multiple times.
+
+To safeguard users' staked assets from hackers, this SIP introduces a new non-transferable `NativeStakedSui` object to encapsulate `StakedSui`, which can be utilized in native stake scenarios, enhancing user security.
+
+## Specification
+
+Added encapsulation of the `StakedSui` object with the `NativeStakedSui` object, where `NativeStakedSui` only possesses the `key` capability and is non-transferable. 
+Added corresponding `add` and `withdraw` functions for native stake.
+- request_add_native_stake
+- request_add_native_stake_non_entry
+- request_withdraw_native_stake
+- request_withdraw_native_stake_non_entry
+
+
+### What Changed
+
+The new code is located in the [sui_system.move](https://github.com/MystenLabs/sui/tree/main/crates/sui-framework/packages/sui-system/sources/sui_system.move) module.
+
+Newly added data structure.
+
+```rust
+public struct NativeStakedSui has key {
+    id: UID,
+    staked_sui: StakedSui,
+}
+```
+
+Newly added functions.
+
+```rust
+/// Add native stake to a validator's staking pool.
+public entry fun request_add_native_stake(
+    wrapper: &mut SuiSystemState,
+    stake: Coin<SUI>,
+    validator_address: address,
+    ctx: &mut TxContext,
+) {
+    let native_staked_sui = request_add_native_stake_non_entry(wrapper, stake, validator_address, ctx);
+    transfer::transfer(native_staked_sui, ctx.sender());
+}
+
+/// The non-entry version of `request_add_native_stake`, which returns the native staked SUI instead of transferring it to the sender.
+public fun request_add_native_stake_non_entry(
+    wrapper: &mut SuiSystemState,
+    stake: Coin<SUI>,
+    validator_address: address,
+    ctx: &mut TxContext,
+): NativeStakedSui {
+    let staked_sui = request_add_stake_non_entry(wrapper, stake,validator_address, ctx);
+    NativeStakedSui {
+        id: object::new(ctx),
+        staked_sui,
+    }
+}
+
+/// Withdraw native stake from a validator's staking pool.
+public entry fun request_withdraw_native_stake(
+    wrapper: &mut SuiSystemState,
+    native_staked_sui: NativeStakedSui,
+    ctx: &mut TxContext,
+) {
+    let withdrawn_stake = request_withdraw_native_stake_non_entry(wrapper, native_staked_sui, ctx);
+    transfer::public_transfer(withdrawn_stake.into_coin(ctx), ctx.sender());
+}
+
+/// Non-entry version of `request_withdraw_native_stake` that returns the withdrawn SUI instead of transferring it to the sender.
+public fun request_withdraw_native_stake_non_entry(
+    wrapper: &mut SuiSystemState,
+    native_staked_sui: NativeStakedSui,
+    ctx: &mut TxContext,
+) : Balance<SUI> {
+    let NativeStakedSui { id, staked_sui } = native_staked_sui;
+    object::delete(id);
+    let self = load_system_state_mut(wrapper);
+    self.request_withdraw_stake(staked_sui, ctx)
+}
+```
```

Comment:
Given the motivations provided in the `Motivation` section I was a bit surprised to find these non-entry public functions as a part of the proposed API because they fully re-introduce the exact same issue that is trying to be addressed by this SIP. With a function like this one, even if the `NativeStakedSui` is non-transferable, if an attacker can trick a user into signing a single transaction, that transaction could have the following comands:

1. call `request_withdraw_native_stake_non_entry` with any and all `NativeStakedSui` found in the user's account
2. pass the results of all of those calls to the native `transfer_objects` command, sending all proceeds directly to an account under the control of the attacker.

Reply:
You've spotted the key point.

Reply:
Thank you for pointing out my mistake. I was writing the SIP in the early morning hours and my mind wasn't very clear.

Reply:
At that time, I was just imitating the function interface of StakedSui, and here I should only retain the entry function.